applications/StructuralMechanicsApplication/custom_elements/base_solid_element.h

```diff
@@ -945,6 +949,49 @@ class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) BaseSolidElement
         }
     }
 
+    /**
+     * @brief This method rotates the F or strain according to local axis from
+     * global to local coordinates
+     * @param rValues The constitutive laws parameters
+     */
+    void RotateToLocalAxes(ConstitutiveLaw::Parameters &rValues);
+
+    /**
+     * @brief This method rotates the F or strain according to local axis from
+     * local de global
+     * @param rValues The constitutive laws parameters
+     */
+    void RotateToGlobalAxes(ConstitutiveLaw::Parameters &rValues);
+
+    /**
+     * @brief This method builds the rotation matrices and local axes
+     */
+    void BuildRotationSystem(
+        BoundedMatrix<double, 3, 3> &rRotationMatrix,
+        const SizeType StrainSize);
+
+    /**
+     * @brief This method checks is an element has to be rotated
+     * according to a set of local axes
+     */
+    bool IsElementRotated()
```

Comment:
```suggestion
    bool IsElementRotated() const
```

and please move the implementation to the cpp